android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLView.java

```diff
@@ -0,0 +1,656 @@
+        _mapOptions.attributionEnabled(!value);
+        if (_map != null) {
+            _map.getUiSettings().setAttributionEnabled(!value);
+        }
+    }
+
+    public void setLogoIsHidden(boolean value) {
+        if (_mapOptions.getLogoEnabled() == !value) { return; }
+        _mapOptions.logoEnabled(!value);
+        if (_map != null) {
+            _map.getUiSettings().setLogoEnabled(!value);
+        }
+    }
+
+    public void setCompassIsHidden(boolean value) {
+        if (_mapOptions.getCompassEnabled() == !value) { return; }
+        _mapOptions.compassEnabled(!value);
+        if (_map != null) {
+            _map.getUiSettings().setCompassEnabled(!value);
+        }
+    }
+
+    public void setContentInset(int top, int right, int bottom, int left) {
+        if (top == _paddingTop &&
+            bottom == _paddingBottom &&
+            left == _paddingLeft &&
+            right == _paddingRight) { return; }
+        _paddingTop = top;
+        _paddingRight = right;
+        _paddingBottom = bottom;
+        _paddingLeft = left;
+        if (_map != null) { _map.setPadding(left, top, right, bottom); }
+    }
+
+    // Events
+
+    void emitEvent(String name, @Nullable WritableMap event) {
+        if (event == null) {
+            event = Arguments.createMap();
+        }
+        ((ReactContext)getContext())
+                .getJSModule(RCTEventEmitter.class)
+                .receiveEvent(getId(), name, event);
+    }
+
+    WritableMap serializePoint(LatLng point) {
+        PointF screenCoords = _map.getProjection().toScreenLocation(point);
+
+        WritableMap event = Arguments.createMap();
+        WritableMap src = Arguments.createMap();
+        src.putDouble("latitude", point.getLatitude());
+        src.putDouble("longitude", point.getLongitude());
+        src.putDouble("screenCoordX", screenCoords.x);
+        src.putDouble("screenCoordY", screenCoords.y);
+        event.putMap("src", src);
+        return event;
+    }
+
+    @Override
+    public void onMapClick(LatLng point) {
+        emitEvent("onTap", serializePoint(point));
+    }
+
+    @Override
+    public void onMapLongClick(@NonNull LatLng point) {
+        emitEvent("onLongPress", serializePoint(point));
+    }
+
+    @Override
+    public void onMyLocationChange(@Nullable Location location) {
+        WritableMap event = Arguments.createMap();
+        WritableMap src = Arguments.createMap();
+
+        if (location == null) {
+            src.putString("message", "Could not get user location");
+            event.putMap("src", src);
+            emitEvent("onLocateUserFailed", event);
+            return;
+        }
+
+        src.putDouble("latitude", location.getLatitude());
+        src.putDouble("longitude", location.getLongitude());
+        src.putDouble("magneticHeading", location.getBearing());
+        src.putDouble("trueHeading", location.getBearing());
+        src.putDouble("headingAccuracy", 10.0);
```

Comment:
Yeah. We could expose accurracy on Android separately, but I'd rather not serve meters as degrees.